Lesson1/Lesson1/Rational.cs

```diff
@@ -0,0 +1,167 @@
+﻿public class Rational
+{
+    private int _Numerator; //числитель 
+
+    private int _Denominator; //знаменатель
+
+    public float RationalNumber => ((float)_Numerator) / ((float)_Denominator);
+
+    public override string ToString() => $"{Numerator}/{Denominator}";
+
+    public Rational ()
+    {
+        _Numerator = 1;
+        _Denominator = 1;
+    }
+
+    public Rational (int numerator, int denominator)
+    {
+        if (denominator == 0)
+            throw new Exception("На ноль делить нельзя!");
+        int nod= 1;
+        if (numerator == 0)
+        {
+            numerator = 0;
+            denominator = 1;
+            return;
+        }
+
+        if (numerator > 0)
+            nod= NOD(numerator, denominator);
+        else
+            nod = NOD(numerator, denominator);
+        _Numerator = numerator / nod;
+        _Denominator = denominator / nod;
+    }
+
+    public int Numerator
+    {
+        get => _Numerator; 
+        set => _Numerator = value; 
+    }
+
+    public int Denominator
+    {
+        get => _Denominator;
+        set => _Denominator = value;
+    }
+
+    /// <summary>Статический метод возвращающий наибольший общий делитель двух положительных чисел</summary>
+    /// <param name="a">Первое число</param>
+    /// <param name="b">Второе число</param>
+    /// <returns>Возвращает ноль - если оба чисела равно нулю, в остальных случаях НОД</returns>
+    public static int NOD (int a, int b)
+    {
+        while (b != 0)
+        {
+            int temp = b;
+            b = a % b;
+            a = temp;
+        }
+        return a;
+    }
+
+    public static implicit operator float(Rational r1) => r1.RationalNumber;
```

Comment:
```suggestion
    public static implicit operator double(Rational r1) => r1.RationalNumber;
```
float на практике не используется толком.

Reply:
Только если есть задача экономить память, либо быстродействие.